tests/runtime/test_grpc_server.py

```diff
@@ -1283,6 +1283,25 @@ def test_construct_with_options(open_port, sample_train_service, sample_int_file
             assert context.value.code() == grpc.StatusCode.RESOURCE_EXHAUSTED
 
 
+def test_grpc_server_startup(open_port, sample_train_service, sample_int_file):
+    """Make sure that the server can be booted with config options"""
+    socket_dir = tempfile.TemporaryDirectory()
```

Comment:
You can request `tmp_path` in the test args and pytest will give you a temporary directory: https://docs.pytest.org/en/7.4.x/how-to/tmp_path.html

But if this was meant to be named `socket_dir` for readability that's 👍

Reply:
I didn't know about `tmp_path`! However, when I tried it I started to get socket bind errors. I believe it's because unix sockets are limited to [paths with less than ~100 characters](https://stackoverflow.com/questions/34829600/why-is-the-maximal-path-length-allowed-for-unix-sockets-on-linux-108) and tmp_path gives you a very long dir name.

I did update the `TemporaryDirectory` to use a context manager just to make it a bit cleaner

Reply:
woah, that's wild. TIL!